src/main/java/edu/cornell/kfs/sys/service/impl/BatchFileDirectoryServiceImpl.java

```diff
@@ -0,0 +1,30 @@
+package edu.cornell.kfs.sys.service.impl;
+
+import edu.cornell.kfs.sys.service.BatchFileDirectoryService;
+import edu.cornell.kfs.sys.util.SubDirectoryWalker;
+import org.kuali.rice.core.api.util.KeyValue;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BatchFileDirectoryServiceImpl implements BatchFileDirectoryService {
+
+    @Override
+    public List<KeyValue> buildDirectoryKeyValuesList(List<File> rootDirectories) {
```

Comment:
The reason for my question was essentially I would consider caching an implementation decision, so by that way of thinking my first thought would be to put in the impl.  However, consistency with Rice seems most appropriate, so we should go with that.